segmentation-templates-extension/src/SegmentationTemplate.liquid

```diff
@@ -0,0 +1,52 @@
+{%- if flavor contains "react" -%}
+import {
+  reactExtension,
+  useApi,
+  CustomerSegmentationTemplate,
```

Comment:
We could hold off on shipping until we've determined how we're going to expose the public API

Reply:
I would like to ship this as-is and adjust later because it will make it much easier to top-hat the whole flow. I don't mind creating a follow-up PR once we finalize the public API. I did not include the fields that we do not want to expose (like `icon` and `category` as well as the metafield reference), so it will be a fairly minimal change. :) Probably just the component name!

Reply:
I think it's fine as long as the option to generate a segmentation template isn't publicly available.